be/src/formats/parquet/complex_column_reader.cpp

```diff
@@ -105,6 +105,34 @@ Status ListColumnReader::read_range(const Range<uint64_t>& range, const Filter*
     return Status::OK();
 }
 
+Status ListColumnReader::fill_dst_column(ColumnPtr& dst, ColumnPtr& src) {
+    ArrayColumn* array_column_src = nullptr;
+    ArrayColumn* array_column_dst = nullptr;
+    if (src->is_nullable()) {
+        NullableColumn* nullable_column_src = down_cast<NullableColumn*>(src.get());
+        DCHECK(nullable_column_src->mutable_data_column()->is_array());
+        array_column_src = down_cast<ArrayColumn*>(nullable_column_src->mutable_data_column());
+        NullColumn* null_column_src = nullable_column_src->mutable_null_column();
+        NullableColumn* nullable_column_dst = down_cast<NullableColumn*>(dst.get());
+        DCHECK(nullable_column_dst->mutable_data_column()->is_array());
+        array_column_dst = down_cast<ArrayColumn*>(nullable_column_dst->mutable_data_column());
+        NullColumn* null_column_dst = nullable_column_dst->mutable_null_column();
+        null_column_dst->swap_column(*null_column_src);
+        nullable_column_src->update_has_null();
```

Comment:
Why not directly swap the flag of src_nullable_column and dst_nullable_column?

Reply:
for data column, we fill it with reader

Reply:
will do it.